srcpkgs/wezterm/template

```diff
@@ -0,0 +1,51 @@
+# Template file for 'wezterm'
+pkgname=wezterm
+version=20210405.110924.a5bb5be8
+revision=1
+_version=${version//./-}
+wrksrc="${pkgname}-${_version}"
+build_style=cargo
+hostmakedepends="pkgconf python3"
+makedepends="fontconfig-devel libxcb-devel xcb-util-image-devel xcb-util-keysyms-devel
+ xcb-util-wm-devel libxkbcommon-devel libX11-devel wayland-devel openssl-devel zlib-devel
+ harfbuzz-devel freetype-devel libpng-devel"
+depends="libxkbcommon-x11 hicolor-icon-theme"
+short_desc="GPU-accelerated cross-platform terminal emulator and multiplexer"
+maintainer="Artur Sinila <freesoftware@logarithmus.dev>"
+license="MIT"
+homepage="https://github.com/wez/wezterm"
+changelog="https://raw.githubusercontent.com/wez/wezterm/main/docs/changelog.md"
+distfiles="${homepage}/archive/refs/tags/${_version}.tar.gz"
+checksum=199b884b2fbcc3d9e8aa6cd3bd04a4bc01fccecc093b1db3807d98f580a9e4ec
+
+# Use system harfbuzz, freetype, libz & libpng instead of the vendored ones
+export WEZTERM_SYSDEPS=1
+
+pre_build() {
+	# Fixed in rust 1.49, but we still use 1.48
+	# Can be dropped after merging https://github.com/void-linux/void-packages/pull/30259
+	cargo update -p spinning_top --precise 0.2.2
+	# Before 1.1.3, libz-sys used vendored zlib for musl targets by default.
+	# `wezterm` still uses libz-sys-1.1.2. https://github.com/wez/wezterm/pull/754 fixed that,
+	# but new wezterm release hasn't been made yet
+	# TLDR; can be dropped after new `wezterm` is released
+	cargo update -p libz-sys
+}
+
+do_install() {
+	vbin target/${RUST_TARGET}/release/wezterm
+	vbin target/${RUST_TARGET}/release/wezterm-gui
+	vbin target/${RUST_TARGET}/release/wezterm-mux-server
+	vmkdir usr/share/wezterm/colors
+	vmkdir usr/share/applications/
+	vmkdir usr/share/icons/hicolor/128x128/apps/
+	vmkdir usr/share/icons/hicolor/scalable/apps/
+	install -m 644 assets/colors/* ${DESTDIR}/usr/share/wezterm/colors/
+	vinstall assets/wezterm.desktop 644 usr/share/applications/
+	vinstall assets/icon/terminal.png 644 usr/share/icons/hicolor/128x128/apps/ wezterm.png
+	vinstall assets/icon/wezterm-icon.svg 644 usr/share/icons/hicolor/scalable/apps/ wezterm.svg
+}
+
+post_install() {
+	vlicense LICENSE.md
+}
```

Comment:
No need to have post install as its own thing.

That said, not sure how I feel about yet another GPU accelerated terminal.

Reply:
> That said, not sure how I feel about yet another GPU accelerated terminal.

You've got a point!  But I'll vote in favor of having it because it:
- serves a different purpose than existing GPU-accelerated terminals (it _has_ GPU acceleration, but that doesn't appear to be its first priority--instead, its priority is adding more features, e.g. better support for emoji, tabbing/multiplexing, native SSH & serial clients, etc.)
- seems kind of cool & is fairly well-established at this point (1.7k stars).  I would enjoy having it & being able to play around w/ it.

Just my 2c :)

Reply:
> No need to have post install as its own thing.

Do you want me to move `vlicense` to the `do_install` function?

> That said, not sure how I feel about yet another GPU accelerated terminal.

`alacritty` doesn't support ligatures. `kitty` supports them, but it's half-written in `python`, an interpreted language.
`wezterm` supports ligatures & is written in Rust.
Yes, `wezterm`'s has ~550 dependencies, thus build time is quite long (~30 minutes), but it's not too bad IMHO.